examples/examples/dragonquotes/dragon_queryable.rs

```diff
@@ -0,0 +1,122 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use async_std::task::sleep;
+use futures::prelude::*;
+use futures::select;
+use rand::{thread_rng, Rng};
+use std::convert::TryFrom;
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::num::ParseIntError;
+use std::path::Path;
+use std::time::Duration;
+use zenoh::config::Config;
+use zenoh::prelude::r#async::AsyncResolve;
+use zenoh::prelude::*;
+
+const QUOTES_INPUT_PATH: &str = "quotes.txt";
+
+const GET_ANY_PATH: &str = "zenoh/quote/any";
+const GET_QUOTE: &str = "zenoh/quote/id/*";
+
+#[async_std::main]
+async fn main() {
+    // initiate logging
+    env_logger::init();
+
+    let get_any_key_expr = KeyExpr::try_from(GET_ANY_PATH).unwrap();
+    let get_quote_key_expr = KeyExpr::try_from(GET_QUOTE).unwrap();
+
+    let quotes: Vec<String> = quotes_from_file(QUOTES_INPUT_PATH);
+
+    println!("Opening session...");
+    let session = zenoh::open(Config::default()).res().await.unwrap();
+
+    println!("Creating Queryable on '{}'...", GET_ANY_PATH);
+    let get_any_queryable = session.declare_queryable(GET_ANY_PATH).res().await.unwrap();
+
+    println!("Creating Queryable on '{}'...", GET_QUOTE);
+    let get_quote_queryable = session.declare_queryable(GET_QUOTE).res().await.unwrap();
+
+    println!("Enter 'q' to quit...");
+    let mut stdin = async_std::io::stdin();
+    let mut input = [0_u8];
+    loop {
+        select!(
+            query = get_any_queryable.recv_async() => {
+                let query = query.unwrap();
+                println!(">> [Queryable ] Received Query '{}'", query.selector());
+                let random_quote = get_random_quote(&quotes);
+                query.reply(Ok(Sample::new(get_any_key_expr.clone(), random_quote))).res().await.unwrap();
+            },
+
+            query = get_quote_queryable.recv_async() => {
+                let query = query.unwrap();
+                println!(">> [Queryable ] Received Query '{}'", query.selector());
+                if query.key_expr().is_wild() {
+                    let all_quotes = get_all_quotes(&quotes);
+                    query.reply(Ok(Sample::new(get_quote_key_expr.clone(), all_quotes))).res().await.unwrap();
```

Comment:
Here, the Zenoh idiom would be to loop over the quotes, providing one reply per quote, with each reply using the keyexpr you would have used to get it individually.

Reply:
Ok changed the code and followed your advice. Although if I do a get I'll only receive the first one, I had to use the z_query_sub example to get all the quotes. I guess the idea of the get is to retrieve the last value of a queryable... am I right?

Reply:
You got trolled by the consolidation, because you still replied on `quotes/id/*` rather than each key individually. When you reply with a wild KE, you imply that EVERY ressource matching that KE has that value. If you say that multiple times with different values each time, Zenoh will treat that as "damn, this set of values changed like crazy recently, good thing I only need to keep one value per KE" :)

Reply:
I see, got it, and tackled it. I notice the received replies come unsorted, independently of the order in which I sent them. I guess this is because each reply runs in parallel and the quotes order is not deterministic.